### ISSUE TITLE
Overfit in Val

### ISSUE THREAD

**Issue by Taoboan1999** [ultralytics/ultralytics#20903]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I am currently training the **YOLOv12** model using my own dataset, and on the map50 data in the validation set, I often see the following changes in results: 

0.67 → 0.68 → 0.69 → 0.70 → 0.71 → 0.68 → 0.67 → 0.67 → 0.68 → 0.67 → 0.67. 

I have ensured that the maximum batch size is 8 and the image size is maximized. I have tried adjusting the learning rate lr0 to 0.001 → 0.0001 → 0.00001 and increased the data augmentation parameters (mosaic). 

Since I cannot modify all the data in the training and validation sets, how can I improve this unfavorable situation? Additionally, I have not yet attempted to modify the parameters (box, cls, dfl). Could this be related to the settings of these three parameters? 

Below are my current parameter settings: 

lr0=0.001, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=5.0, cls=0.1, d fl=1.0, val=True, hsv_h=0.0, hsv_s=0.0, hsv_v=0.0, degrees=0.0, translate=0.0, scale=0.2, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=0.7, mixup=0.5, cutmix=0. 7

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Taoboan1999, thank you for reaching out and for your interest in Ultralytics 🚀! This is an automated response to help guide you while an Ultralytics engineer reviews your question and will assist you soon.

We recommend visiting the [Docs](https://docs.ultralytics.com/) where you’ll find detailed [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, as well as answers to many common training and validation questions.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

For custom training ❓ questions like this, please include as much detail as possible—dataset info, image samples, and full training logs can be very helpful. Also, check that you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best! For real-time chat, hop into [Discord](https://discord.com/invite/ultralytics) 🎧. For more in-depth discussions, check out [Discourse](https://community.ultralytics.com/). Or join the conversation on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge and get feedback from others.

## Upgrade

Upgrade to the latest `ultralytics` package, including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to make sure your issue is not already resolved in the latest release:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of the following up-to-date, verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for your patience and for contributing to Ultralytics!

**Comment by glenn-jocher** [comment]
The fluctuating validation mAP50 you're seeing (peaking then dropping) is a classic sign of overfitting. I'd recommend switching from YOLOv12 to [YOLO11](https://docs.ultralytics.com/models/yolo11/) as YOLO12 suffers from training instability and is not recommended for production use.

Your current loss weights are significantly lower than defaults - try increasing `cls=0.5` (you have 0.1) and `box=7.5` (you have 5.0) as per the [default training configuration](https://docs.ultralytics.com/usage/cfg/). Also consider adding `dropout=0.1` for regularization and reducing your learning rate further to `lr0=0.0001` with longer training to allow for more stable convergence.

**Comment by Taoboan1999** [comment]
> The fluctuating validation mAP50 you're seeing (peaking then dropping) is a classic sign of overfitting. I'd recommend switching from YOLOv12 to [YOLO11](https://docs.ultralytics.com/models/yolo11/) as YOLO12 suffers from training instability and is not recommended for production use.
> 
> Your current loss weights are significantly lower than defaults - try increasing `cls=0.5` (you have 0.1) and `box=7.5` (you have 5.0) as per the [default training configuration](https://docs.ultralytics.com/usage/cfg/). Also consider adding `dropout=0.1` for regularization and reducing your learning rate further to `lr0=0.0001` with longer training to allow for more stable convergence.

![Image](https://github.com/user-attachments/assets/59ab0691-943e-4b89-bb24-7e2baba2000c)

![Image](https://github.com/user-attachments/assets/60feee3f-b08e-4847-84e2-d6050c943392)

I see that dropout is only used in classify. Would dropout be effective in detect?

**Comment by glenn-jocher** [comment]
Yes, dropout is effective for detection tasks too. While the [configuration documentation](https://docs.ultralytics.com/usage/cfg/) mentions it's for "classification tasks," dropout is a general regularization technique that helps prevent overfitting in any neural network by randomly omitting units during training. Since YOLO detection models contain both classification and regression heads, dropout can help regularize the entire network and reduce overfitting in your detection results.
